src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,31 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int USER_MINIMUM_AGE = 18;
+    public static final int PASSWORD_MINIMUM_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null || user.getLogin() == null || user.getPassword() == null
+                || user.getAge() == null) {
+            throw new RegistrationServiceException("User fields can't be null/empty/negative");
+        }
+        if (user.getAge() < 0 || user.getLogin().isEmpty()) {
+            throw new RegistrationServiceException("User age can't be neither negative nor empty");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationServiceException("User with login "
+                    + user.getLogin() + " exists");
+        }
+        if (user.getAge() < USER_MINIMUM_AGE || user.getAge() > 110) {
```

Comment:
```suggestion
        if (user.getAge() < USER_MINIMUM_AGE) {
```
we don't have max age